BLSQ/iaso
# WC2-196/ WC2-197: Workflows follow-ups - current date and current date time, calculate operators

The follow-up feature should allow to pick a date field and to compare it to the “now” variable that represents the current moment. 
On top of “enter date”, it should be possible to pick as option “current time” for date fields, so that on mobile, we can compare the current time with. This would result in using the “currentTime” string in the formula at the place where the date was entered.

Also calculate fields should support `<=` and `>=` operators

Related JIRA tickets : [WC2-196](https://bluesquare.atlassian.net/browse/WC2-196)  [WC2-197](https://bluesquare.atlassian.net/browse/WC2-197)

## Self proofreading checklist


## Changes

- adding custom fields for query builder (currentDateTime, currentDate), make it optional 
- create custom component for date time
- create custom component for new query builder fields 
- support operators for calculate fields

## How to test

- Go to beneficiary types
- Select a type with a reference form using date and date time 
[registration_sudan_2023030845465465.xlsx](https://github.com/BLSQ/bluesquare-components/files/11219129/registration_sudan_2023030845465465.xlsx)
- select / create a version and add a follow-up
- you should be able to select current date or date time for date time and date fields
- those custom fields should not be available while performing a search in the 
- calculate fields should support  `<=` and `>=` operators


## Print screen / video

https://user-images.githubusercontent.com/12494624/231686623-d79538be-6bb2-4e88-9d34-03a738f64192.mov

<img width="1010" alt="Screenshot 2023-04-13 at 11 22 24" src="https://user-images.githubusercontent.com/12494624/231716523-69012f41-502f-46af-b527-db5ecd1ce6e5.png">


[WC2-196]: https://bluesquare.atlassian.net/browse/WC2-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


## Notes

⚠️ Please use/merge [this](https://github.com/BLSQ/bluesquare-components/pull/113) before ⚠️ 


[WC2-197]: https://bluesquare.atlassian.net/browse/WC2-197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
